frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -333,22 +333,24 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 				self,
 				origin: Self::RuntimeOrigin
 			) -> #frame_support::dispatch::DispatchResultWithPostInfo {
-				match self {
-					#(
-						Self::#fn_name { #( #args_name_pattern, )* } => {
-							#frame_support::sp_tracing::enter_span!(
-								#frame_support::sp_tracing::trace_span!(stringify!(#fn_name))
-							);
-							#maybe_allow_attrs
-							<#pallet_ident<#type_use_gen>>::#fn_name(origin, #( #args_name, )* )
-								.map(Into::into).map_err(Into::into)
+				#frame_support::dispatch_context::run_in_context(|| {
```

Comment:
This could also be a macro expansion as optimization instead of a closure+function call every time. But not sure if its worth it.

Reply:
Then I would need to make `DISPATCH_CONTEXT` public which I don't really want. However, I don't think that this adds any significant overhead. We don't call `dispatch` in any tight loop where it could be useful to have this optimization.